packages/api/src/services/contacts.service.ts

```diff
@@ -15,19 +16,38 @@ export class ContactsService {
         return this.contactRepository.find({ where: { userId }, order: { contactName: "ASC" } });
     }
 
+    async findSpecific(userId: number, contactName: string) {
+        return this.contactRepository.find({ where: { userId, contactName } });
+    }
+
+    async delete(userId: number, contactName: string) {
+        return this.contactRepository.delete({ userId: userId, contactName: contactName });
+    }
+    
     async create(createContactRequest: CreateContactRequest): Promise<Contact[]> {
         const { userId, contactName } = createContactRequest;
 
-        const walletAddrs = [...createContactRequest.btcWallets, ...createContactRequest.ethWallets];
-        const contacts = walletAddrs.map((addr) =>
+        const walletAddrsDelete = [...createContactRequest.btcWalletsDelete, ...createContactRequest.ethWalletsDelete];
+        const contactsToDelete = walletAddrsDelete.map((addr) =>
+            this.contactRepository.create({
+                userId,
+                contactName,
+                walletAddress: addr,
+            }),
+        );
+
+        await this.contactRepository.remove(contactsToDelete);
+
+        const walletAddrsInsert = [...createContactRequest.btcWallets, ...createContactRequest.ethWallets];
+        const contactsToAdd = walletAddrsInsert.map((addr) =>
             this.contactRepository.create({
                 userId,
                 contactName,
                 walletAddress: addr,
             }),
         );
 
-        await this.contactRepository.insert(contacts);
-        return contacts;
+        await this.contactRepository.insert(contactsToAdd);
+        return contactsToAdd;
```

Comment:
Just to make it uniform

```suggestion
        return [...contactsToAdd, ...contactsToDelete];
```